pallets/automation-time/src/tests.rs

```diff
@@ -289,6 +293,33 @@ fn schedule_native_transfer_works() {
 	})
 }
 
+#[test]
+fn calculate_schedule_fee_amount_works() {
+	new_test_ext(START_BLOCK_TIME).execute_with(|| {
+		let destination = MultiLocation::new(1, X1(Parachain(PARA_ID)));
+		let delegator_account = AccountId32::new(DELEGATOR_ACCOUNT);
+		let execution_count = 2;
+
+		let action: ActionOf<Test> = Action::XCMP {
+			destination,
+			schedule_fee: destination.into(),
+			execution_fee: AssetPayment { asset_location: destination.into(), amount: 100 },
+			encoded_call: vec![3, 4, 5],
+			encoded_call_weight: Weight::from_ref_time(100_000),
+			overall_weight: Weight::from_ref_time(200_000),
+			schedule_as: Some(delegator_account),
+			flow: InstructionSequence::PayThroughRemoteDerivativeAccount,
+		};
+		let schedule_fee_amount =
+			AutomationTime::calculate_schedule_fee_amount(&action, execution_count)
+				.expect("Calculate schedule fee amount should work");
+		let expected_schedule_fee_amount =
+			FEE_PER_SECOND * XCMP_TASK_REF_TIME * (execution_count as u128) /
```

Comment:
`action` is what you passed in to the function calculate_schedule_fee_amount. You need to use the variable values within **action** for calculation, instead of using a separate const variable XCMP_TASK_REF_TIME. XCMP_TASK_REF_TIME is not used in mock.

Reply:
WEIGHT_REF_TIME_PER_SECOND is a global variable for our chain, so that’s the only constant variable which can be used in this line of validation.

Fee per second is in relation to each destination, you need to mock and  to get them by destination.fee_per_second.

Reply:
If the task is a xcmp task. the weight of the task is a const. 

It is defined here.

https://github.com/OAK-Foundation/OAK-blockchain/blob/69464509e9dd628485b6633df1b33718ce42cbab/pallets/automation-time/src/weights.rs#L311

It is not calculated based on the data of the task/action.

So, I think it should be fine to define the XCMP_TASK_REF_TIME constant as well.